eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelay.scala

```diff
@@ -112,29 +114,68 @@ class ChannelRelay private(nodeParams: NodeParams,
   private val forwardFailureAdapter = context.messageAdapter[Register.ForwardFailure[CMD_ADD_HTLC]](WrappedForwardFailure)
   private val addResponseAdapter = context.messageAdapter[CommandResponse[CMD_ADD_HTLC]](WrappedAddResponse)
 
+  private val nextBlindingKey_opt = r.payload match {
+    case payload: IntermediatePayload.ChannelRelay.Blinded => Some(payload.nextBlinding)
+    case _: IntermediatePayload.ChannelRelay.Standard => None
+  }
+
+  /** Channel id explicitly requested in the onion payload. */
+  private val requestedChannelId_opt = r.payload.outgoing match {
+    case Left(_) => None
+    case Right(outgoingChannelId) => channels.collectFirst {
+      case (channelId, channel) if channel.shortIds.localAlias == outgoingChannelId => channelId
+      case (channelId, channel) if channel.shortIds.real.toOption.contains(outgoingChannelId) => channelId
+    }
+  }
+
   private case class PreviouslyTried(channelId: ByteVector32, failure: RES_ADD_FAILED[ChannelException])
 
-  def relay(previousFailures: Seq[PreviouslyTried]): Behavior[Command] = {
+  def start(): Behavior[Command] = {
+    r.payload.outgoing match {
+      case Left(walletNodeId) => wakeUp(walletNodeId)
+      case Right(requestedShortChannelId) =>
```

Comment:
Note how this can be extended to detect "custom" scids and insert the wake-up step, by resolving the `walletNodeId` using our push notifications DB (or an existing channel if we have one, which avoids a DB call).